File: FamilyVacationPlanner/src/Itineraries.Core/Models/Itinerary.cs
namespace Itineraries.Core.Models;

public class Itinerary
{
    public Guid ItineraryId { get; set; }
    public Guid TenantId { get; set; }
    public Guid TripId { get; set; }
    public DateTime Date { get; set; }
    public string? Activity { get; set; }
    public string? Location { get; set; }
    public DateTime CreatedAt { get; set; } = DateTime.UtcNow;
}

File: FamilyVacationPlanner/src/FamilyVacationPlanner.WebApp/projects/family-vacation-planner-shared/src/lib/models/itinerary.ts
export interface Itinerary {
  itineraryId: string;
  tripId: string;
  date: string;
  activity?: string;
  location?: string;
  createdAt: string;
}

export interface CreateItineraryCommand {
  tripId: string;
  date: string;
  activity?: string;
  location?: string;
}

export interface UpdateItineraryCommand {
  itineraryId: string;
  tripId: string;
  date: string;
  activity?: string;
  location?: string;
}
